### ISSUE TITLE
Editor scrollbar wildly jittering on drag.

### ISSUE THREAD

**Issue by avencherus** [godotengine/godot#23314]
Win10 64bit commit: eeee47196c19dc271d6fcf8981946a1a29efd16a + https://github.com/godotengine/godot/pull/23059

I puzzled over this one for a bit, but I cannot reproduce it, and have only seen it once.  It happened at least from returning focus to the editor from an outside window.

Anyone else run into this?  

_A video that shows the jittering better:_ https://streamable.com/7vvjs

![godot_scroll_jitter](https://user-images.githubusercontent.com/13004169/47590102-00f09280-d974-11e8-9eb8-3158335e2974.gif)

**Comment by spongeboburu** [comment]
I can confirm as this has happened to me as well. However, in my case I couldn't make it stop jittering so a restart was needed since closing the text editor was not enough. It was also isolated to that one file, opening the text editor for another file it was fine.

**Comment by avencherus** [comment]
Yeah I also had to restart to free it up.

**Comment by somnivore** [comment]
I've encountered this issue as well, trying to select some code but missing and clicking into the scrollbar instead.

1. Left click and hold in the empty space above or below the scroll bar
2. Drag the mouse away
3. The scrollbar will now not want to move to any other position but the last position it scrolled to.

Middle-mouse scrolling now does not work.  You can drag the scrollbar, but it will jitter, and try to return to that one location.  You can continue to move the scrollbar up or down by clicking in the empty space above or below it, but the behavior otherwise will continue.

To fix it, you can continue to click in the empty space above the scroll bar until it reaches the top, where it will then return to normal--or restart the editor.

**Comment by bojidar-bg** [comment]
Tested with a142c9a2f0be65218058c35b94202fbc4fd2c84c (original smooth scrolling commit), couldn't reproduce.
So, I ran a bisect:
```
git bisect start
# bad: [2d6b994e476f00e5725117501f6fc006eadd0d30] Massive speed up on deleting nodes, fixes #18673 Also makes the editor exit faster
git bisect bad 2d6b994e476f00e5725117501f6fc006eadd0d30
# good: [a142c9a2f0be65218058c35b94202fbc4fd2c84c] Added smooth scrolling to TextEdit
git bisect good a142c9a2f0be65218058c35b94202fbc4fd2c84c
# good: [e3eb6869068edb2617a311f89e93b92a9aa5b905] Tree now uses UI actions instead of keys
git bisect good e3eb6869068edb2617a311f89e93b92a9aa5b905
# good: [274d1bc683bbd7f31b323f217bbe4512a8c275a1] Merge pull request #20233 from willnationsdev/gsc-editor
git bisect good 274d1bc683bbd7f31b323f217bbe4512a8c275a1
# good: [726e836cd9b6eee5e060b694e076535ad5b5a7ae] GDScript: Fix infinite loop in autocomplete
git bisect good 726e836cd9b6eee5e060b694e076535ad5b5a7ae
# bad: [43c28681add010b5daccbcfde3e3065d7be7b52c] Merge pull request #22939 from merumelu/tileset-add-texture-fix
git bisect bad 43c28681add010b5daccbcfde3e3065d7be7b52c
# bad: [22c3995a04c59a24c109db029f2492253c494d88] Merge pull request #22576 from DualMatrix/multi_edit_exit
git bisect bad 22c3995a04c59a24c109db029f2492253c494d88
# good: [aaef640b8c3e7c7c49aae11a1c8c9e606223fefd] Merge pull request #22383 from Faless/ws_close
git bisect good aaef640b8c3e7c7c49aae11a1c8c9e606223fefd
# bad: [d8af4f9e02277194ca1bb053447a25bfed2b6524] Merge pull request #22506 from akien-mga/thirdparty-no-warnings
git bisect bad d8af4f9e02277194ca1bb053447a25bfed2b6524
# good: [bca2d3ad40da1b5bea2e55a8640615d21736b175] Fix invalid comparison warnings: [-Wbool-compare] and [-Wenum-compare]
git bisect good bca2d3ad40da1b5bea2e55a8640615d21736b175
# good: [1c60465c5958446a2daccc54a1c58c3dee9b21a3] Merge pull request #22422 from Faless/server_dummy_vr
git bisect good 1c60465c5958446a2daccc54a1c58c3dee9b21a3
# bad: [d55491a7b8093bfadd257bd9df6a04f76e05d8ca] Merge pull request #22407 from DualMatrix/step_int
git bisect bad d55491a7b8093bfadd257bd9df6a04f76e05d8ca
# good: [f87fb2c795f93fa4171c8ab0b6a5af49b013dcd7] Merge pull request #22413 from lupoDharkael/array-mesh-docs
git bisect good f87fb2c795f93fa4171c8ab0b6a5af49b013dcd7
# bad: [a5017f925f32c0bf8730f50c08ffc8d4ec1b7794] Merge pull request #22405 from DualMatrix/better_step
git bisect bad a5017f925f32c0bf8730f50c08ffc8d4ec1b7794
# bad: [98395f652c285a4c661a634a98fb9d63345c8a7b] Fixed step value not being respected for ranges in some cases.
git bisect bad 98395f652c285a4c661a634a98fb9d63345c8a7b
# first bad commit: [98395f652c285a4c661a634a98fb9d63345c8a7b] Fixed step value not being respected for ranges in some cases.
```
Seems like #22405 is the culprit. CC @DualMatrix.
Also, CC @poke1024 and @Paulb23, since they worked on smooth scrolling in the past.

**Comment by DualMatrix** [comment]
I assume that somewhere a range with a step is being used and now it tries to snap to the step which it wouldn't do previously. Should be a simple fix by just removing the step from the range thats the culprit
